apps/asap-server/src/utils/instrumented-framework.ts

```diff
@@ -30,10 +31,15 @@ if (lightstepToken && NODE_ENV !== 'test') {
 
 export const http = (
   fn: (request: lambda.Request) => Promise<lambda.Response>,
+  logger?: PinoLambdaLogger,
 ) => async (
   event: APIGatewayProxyEventV2,
   context?: Context,
 ): Promise<APIGatewayProxyResultV2> => {
+  if (logger) {
+    // casting as non-null as the lambda always gives context
+    logger.withRequest(event, context!);
```

Comment:
The parameter is optional. Could you change that? Because it doesn't look great to have an explicit optional param and then saying it's non-null 😄 Either way, you'd need an eslint rule to be able to get this through.

Reply:
yeah I changed it just now. To be fair I don't think lambda ever omits these and I think the types are simply invalid. I could change it but then every test using the handler would have to attach context (which is fine imo)